src/librustc_typeck/variance/constraints.rs

```diff
@@ -161,12 +161,7 @@ impl<'a, 'tcx> ConstraintContext<'a, 'tcx> {
                 self.add_constraints_from_sig(current_item, tcx.fn_sig(def_id), self.covariant);
             }
 
-            _ => {
-                span_bug!(
-                    tcx.def_span(def_id),
-                    "`build_constraints_for_item` unsupported for this item"
-                );
-            }
+            _ => {}
```

Comment:
It is still useful to emit a bug if we encounter other types here, for example `bool` or `ty::Param`.

My idea was to keep that catch arm and add a separate branch for `ty::Error(_)`